sim/common/mop/cloaks_phase_4_54.go

```diff
@@ -0,0 +1,215 @@
+package mop
+
+import (
+	"math"
+	"time"
+
+	"github.com/wowsims/mop/sim/common/shared"
+	"github.com/wowsims/mop/sim/core"
+	"github.com/wowsims/mop/sim/core/proto"
+)
+
+type cloakConfig struct {
+	label  string
+	itemID int32
+}
+
+func init() {
+	// Xing-Ho, Breath of Yu'lon
+	// Your damaging spells have a chance to grant 100% critical strike chance for 4 sec.  (Approximately 0.62 procs per minute)
+	core.NewItemEffect(102246, func(agent core.Agent, state proto.ItemLevelState) {
+		character := agent.GetCharacter()
+		label := "Xing-Ho, Breath of Yu'lon"
+
+		aura := core.MakePermanent(character.RegisterAura(core.Aura{
+			Label: label,
+		}))
+
+		shared.RegisterIgniteEffect(&character.Unit, shared.IgniteConfig{
+			ActionID:           core.ActionID{SpellID: 146198},
+			SpellSchool:        core.SpellSchoolNature,
+			DisableCastMetrics: true,
+			DotAuraLabel:       "Essence of Yu'lon",
+			TickLength:         1 * time.Second,
+			NumberOfTicks:      4,
+			NumberAOEOfTargets: 5,
+			ParentAura:         aura,
+
+			ProcTrigger: core.ProcTrigger{
+				Name:    label,
+				Harmful: true,
+				DPM: character.NewRPPMProcManager(102246, false, core.ProcMaskSpecial|core.ProcMaskProc, core.RPPMConfig{
+					PPM: 2.61100006104,
+				}.WithHasteMod().
+					WithSpecMod(0.25, proto.Spec_SpecArcaneMage).
+					WithSpecMod(0.20000000298, proto.Spec_SpecFireMage).
+					WithSpecMod(0.20000000298, proto.Spec_SpecFrostMage).
+					WithSpecMod(-0.75, proto.Spec_SpecProtectionPaladin).
+					WithSpecMod(-0.75, proto.Spec_SpecProtectionWarrior).
+					WithSpecMod(0.10000000149, proto.Spec_SpecBalanceDruid).
+					WithSpecMod(-0.75, proto.Spec_SpecGuardianDruid).
+					WithSpecMod(-0.75, proto.Spec_SpecBloodDeathKnight).
+					WithSpecMod(0, proto.Spec_SpecShadowPriest).
+					WithSpecMod(0.05000000075, proto.Spec_SpecElementalShaman).
+					WithSpecMod(0.10000000149, proto.Spec_SpecAfflictionWarlock).
+					WithSpecMod(0.25, proto.Spec_SpecDemonologyWarlock).
+					WithSpecMod(0.15000000596, proto.Spec_SpecDestructionWarlock).
+					WithSpecMod(-0.75, proto.Spec_SpecBrewmasterMonk),
+				),
+				Callback: core.CallbackOnSpellHitDealt,
+			},
+
+			DamageCalculator: func(spell *core.Spell, result *core.SpellResult) float64 {
+				return spell.SpellPower() * 2
+			},
+		})
+
+		eligibleSlots := character.ItemSwap.EligibleSlotsForItem(102246)
+		character.ItemSwap.RegisterProcWithSlots(102246, aura, eligibleSlots)
+	})
+
+	newXuenCloakEffect := func(config cloakConfig) {
```

Comment:
Since there are only two inputs here, you can just pass them directly without the need for a config struct.